configure.sh

```diff
@@ -1,29 +1,75 @@
 #!/bin/bash
-if ! type parity > /dev/null; 
-then
+
+# Install parity if this is a linux system
+#
+installLinux() {
     PARITY_DEB_URL=https://parity-downloads-mirror.parity.io/v1.8.6/x86_64-unknown-linux-gnu/parity_1.8.6_amd64.deb
-    if ! type curl > /dev/null; 
+    if ! type curl > /dev/null;
     then
         echo "Installing curl."
         sudo apt-get update -qq
         sudo apt-get install -y -qq curl
-    else 
+    else
         echo "Curl is already installed."
     fi
     echo "Installing parity"
     file=/tmp/parity.deb
     curl -Lk $PARITY_DEB_URL > $file
     sudo dpkg -i $file
     rm $file
-else 
+}
+
+# Install parity if this is a Darwin system (macOS)
+#
+installMac() {
+
+    if  !type brew > /dev/null;
+    then
+
+        echo "This requires Brew."
+        echo "Installing Homebew"
+        /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+    fi
+
+    echo "Installing parity"
+
+    brew tap paritytech/paritytech
+    brew install parity
+}
+
+
+if ! type parity > /dev/null;
+then
+
+    platform='unknown'
+    unamestr=`uname`
+    if [[ "$unamestr" == 'Linux' ]]; then
```

Comment:
Not sure it makes sense to check for "Linux" for apt-specific instructions. Or "Darwin" when it uses Homebrew which I don't think is guaranteed to exist on Mac.

Maybe branch based on the existence of an actual package manager like `apt` or `brew`?

Reply:
Oh I see, your script installs Homebrew. I'd still maybe favour this approach so that installing Homebrew isn't a side effect of this script when it might not be documented/expected. That line also relies on Ruby which may/may not be installed either.

Reply:
ruby is default installed on macOS. I tried using the mac version of the linux install script to install parity from the website, but that does not install the commandline tool to start parity. If you are aware of a better way that I could install it then I can switch it to that but this does the job. Also I can add it to the README that it will install homebrew.

Reply:
Ah okay, ruby being definitely installed helps.

At any rate, my thought was more about instead of switching based on OS, switch based on existence of a package manager. So if `brew` is installed call a function that uses `brew`. If `apt` is installed call one that uses `apt`. If `pacman` or `yum` or whatever is installed eventually we can use that.

Just decouples package manager logic from operating system logic and avoids side-effects in one fell swoop.

Reply:
Thank make sense I can make those changes.